### PR TITLE
Add embedsay and embededit command, Fix bug that caused crash with ed…

### PR DESCRIPTION
**Pull request by mushroom-disaster**
…it commands, Use logging channel from the config in the edit commands

### CODE REVIEW

File: src/bot/commands/edit.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const command = new Command('edit', async (msg, args, context) => {
 	const oldMessage = await context.client.getMessage(channelId, messageId).catch(() => {});
 
 	// log the old content just in case
-	context.client.createMessage(botLogChannelId, oldMessage.content).catch(() => {});
+	if (oldMessage) context.client.createMessage(botLogChannelId, oldMessage.content).catch(() => {});
```
Comment (eritbh) [inline src/bot/commands/edit.js:17]: Nice catch

File: src/bot/commands/embedsay.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	// try to get the channel, if none is found just post it in the channel the message was sent
+	let channel;
+	try {
+		channel = context.client.getChannel(args[0].split('\n')[0]) || await context.client.getRESTChannel(args[0]);
+	} catch (_) {
+		// pass, channel stays undefined
+	}
+
+	if (channel) {
+		const messageContent = args.slice(1, args.length).join(' ');
```
Comment (eritbh) [inline src/bot/commands/embedsay.js:19]: You can omit the second argument to `slice()` to get everything until the end of the array:

```suggestion
		const messageContent = args.slice(1).join(' ');
```

File: src/bot/commands/embedsay.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+		context.client.createMessage(channel.id, {
+			embed: {
+				description: messageContent,
+			},
+		}).catch(() => {});
+	} else {
+		const messageContent = args.slice(0, args.length).join(' ');
+		// same as above
+		if (messageContent.length > 3950) {
```
Comment (eritbh) [inline src/bot/commands/embedsay.js:34]: 4000 is used as the limit above but 3950 here; is that intentional?

File: src/bot/commands/embedsay.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+			return;
+		}
+
+		context.client.createMessage(channel.id, {
+			embed: {
+				description: messageContent,
+			},
+		}).catch(() => {});
+	} else {
+		const messageContent = args.slice(0, args.length).join(' ');
```
Comment (eritbh) [inline src/bot/commands/embedsay.js:32]: This `slice()` call does nothing, I'm pretty sure? It returns a clone of the full array, but we don't need a clone to `join()`, we can do that directly on the original array

```suggestion
		const messageContent = args.join(' ');
```

**Review by eritbh — CHANGES_REQUESTED** [review]

File: src/bot/commands/embedsay.js (hunk 3 above)
Comment (mushroom-disaster) [reply src/bot/commands/embedsay.js:34]: Nope I just forgot to change that. I set them all as 3950 at first but changed them after I saw that the limit was 4090 and not 4000 in the docs

File: src/bot/commands/embedsay.js (hunk 4 above)
Comment (mushroom-disaster) [reply src/bot/commands/embedsay.js]: Yep removed it

File: src/bot/commands/embedsay.js (hunk 2 above)
Comment (mushroom-disaster) [reply src/bot/commands/embedsay.js]: Also removed it

File: src/bot/commands/embededit.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+const botLogChannelId = config.TEMP_loggingChannelID;
+
+const command = new Command('embededit', async (msg, args, context) => {
+	if (args.length < 2) {
+		return context.sendHelp(msg, context);
+	}
+
+	const channelId = args[0];
+	const messageId = args[1];
+	const newMessageContent = args.slice(2, args.length).join(' ');
```
Comment (eritbh) [inline src/bot/commands/embededit.js:13]: One more here

```suggestion
	const newMessageContent = args.slice(2).join(' ');
```

**Review by eritbh — CHANGES_REQUESTED** [review]

**Review by eritbh — APPROVED** [review]

**Pull request merged**